### ISSUE TITLE
test_classparameters.py - wrong search used

### ISSUE THREAD

**Issue by rplevka** [SatelliteQE/robottelo#4018]
[This  PR](https://github.com/SatelliteQE/robottelo/pull/3777) has introduced some class parameters tests.
However, the `setUpClass()` (perhaps the object attributes have changed) fails for all tests:
```bash
cls = <class 'tests.foreman.api.test_classparameters.SmartClassParametersTestCase'>

    @classmethod
    def setUpClass(cls):
        super(SmartClassParametersTestCase, cls).setUpClass()
        cls.puppet_module = "puppetlabs/ntp"
        cls.host_name = settings.server.hostname
        ssh.command(
            'puppet module install --force {0}'.format(cls.puppet_module))
        cls.env = entities.Environment().search(
            query={'search': 'name="production"'}
        )
        if len(cls.env) == 0:
            raise Exception("Environment not found")
        cls.env = cls.env[0]
        cls.proxy = entities.SmartProxy(name=cls.host_name).search()[0]
        cls.proxy.import_puppetclasses(environment=cls.env)
        cls.puppet = entities.PuppetClass().search(
            query={'search': 'name="ntp"'}
>       )[0]
E       IndexError: list index out of range

tests/foreman/api/test_classparameters.py:137: IndexError
```

listing the Puppet classes manually shows why (tl;dr seems like the naming convention changed):
```
$ curl -ku admin:changeme https://sat6com/api/v2/puppetclasses?search="name"~"ntp" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   496    0   496    0     0    629      0 --:--:-- --:--:-- --:--:--   628
{
  "total": 8,
  "subtotal": 3,
  "page": 1,
  "per_page": 20,
  "search": "name~ntp",
  "sort": {
    "by": null,
    "order": null
  },
  "results": {
    "ntp": [
      {
        "id": 3,
        "name": "ntp::config",
        "created_at": "2016-11-23T16:49:35.418Z",
        "updated_at": "2016-11-23T16:49:35.418Z"
      },
      {
        "id": 2,
        "name": "ntp::install",
        "created_at": "2016-11-23T16:49:35.409Z",
        "updated_at": "2016-11-23T16:49:35.409Z"
      },
      {
        "id": 1,
        "name": "ntp::service",
        "created_at": "2016-11-23T16:49:35.390Z",
        "updated_at": "2016-11-23T16:49:35.390Z"
      }
    ]
  }
}

```

fixing the search query to `s/=/~` helps at this step, however `setUpClass` fails further:
```
2016-11-23 17:59:41 - nailgun.client - DEBUG - Making HTTP GET request to https://sat6.com/api/v2/smart_class_parameters with options {'verify': False, 'data': '{"per_page": 100, "search": "puppetclass=\\"ntp\\""}', 'auth': ('admin', 'changeme'), 'headers': {'content-type': 'application/json'}} and no data.
2016-11-23 17:59:42 - nailgun.client - DEBUG - Received HTTP 200 response: {
  "total": 20,
  "subtotal": 0,
  "page": 1,
  "per_page": 100,
  "search": "puppetclass=\"ntp\"",
  "sort": {
    "by": null,
    "order": null
  },
  "results": []
}

E
======================================================================
ERROR: test suite for <class 'tests.foreman.api.test_classparameters.SmartClassParametersTestCase'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/usr/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/usr/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/usr/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/rplevka/work/rplevka/robottelo/tests/foreman/api/test_classparameters.py", line 142, in setUpClass
    raise RuntimeError('There are not enough smart class parameters to'
RuntimeError: There are not enough smart class parameters to work with in provided puppet class

```

again, listing the `smart_class_parameters` shows, that there is none with `puppetclass_name` == 'ntp':

```
$ curl -ku admin:changeme https://sat6.com/api/v2/smart_class_parameters | jq '.results[] .puppetclass_name' | uniq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11034    0 11034    0     0  10542      0 --:--:--  0:00:01 --:--:-- 10548
"access_insights_client"
"foreman_scap_client"
"access_insights_client"
"foreman_scap_client"
"access_insights_client"
"foreman_scap_client"
"access_insights_client"
"foreman_scap_client"
"access_insights_client"
"foreman_scap_client"
"access_insights_client"

```

**Comment by rplevka** [comment]
marking as `urgent` as this one is causing >=33 tests to fail

**Comment by rplevka** [comment]
applies to `CLI` too

**Comment by lpramuk** [comment]
``ntp`` is puppet module name. But here we search for puppetclasses (!= puppet modules)

**Comment by svtkachenko** [comment]
There is also `ntp` class but I think it was deleted in another thread :)
Probably can be closed due to PR #3958

**Comment by svtkachenko** [comment]
Verified with master branch
```python
% nosetests tests/foreman/{api,cli}/test_{classparameters,variables}.py
SSSS.............SSSSSS............................S..S....SSSS........S............SSSSS.SSSS..........SSSSSS........................SSSS....S..S......SSSSS.....S...........S......
----------------------------------------------------------------------
Ran 181 tests in 2599.978s
OK (SKIP=45)
```
